gpflow/likelihoods/base.py

```diff
@@ -48,97 +50,119 @@ def __init__(self, latent_dim: Optional[int], observation_dim: Optional[int]) ->
         :param observation_dim: the dimension of the observation vector Y for a single data point
         """
         super().__init__()
+        self.input_dim = input_dim
         self.latent_dim = latent_dim
         self.observation_dim = observation_dim
 
-    def _check_last_dims_valid(self, F: TensorType, Y: TensorType) -> None:
+    def _check_last_dims_valid(self, X: TensorType, F: TensorType, Y: TensorType) -> None:
         """
         Assert that the dimensions of the latent functions F and the data Y are compatible.
 
-        :param F: function evaluation Tensor, with shape [..., latent_dim]
-        :param Y: observation Tensor, with shape [..., observation_dim]
+        :param X: input tensor, with shape [..., input_dim]
+        :param F: function evaluation tensor, with shape [..., latent_dim]
+        :param Y: observation tensor, with shape [..., observation_dim]
         """
+        self._check_input_dims(X)
         self._check_latent_dims(F)
         self._check_data_dims(Y)
 
-    def _check_return_shape(self, result: TensorType, F: TensorType, Y: TensorType) -> None:
+    def _check_return_shape(
+        self, result: TensorType, X: TensorType, F: TensorType, Y: TensorType
+    ) -> None:
         """
         Check that the shape of a computed statistic of the data
         is the broadcasted shape from F and Y.
 
-        :param result: result Tensor, with shape [...]
-        :param F: function evaluation Tensor, with shape [..., latent_dim]
-        :param Y: observation Tensor, with shape [..., observation_dim]
+        :param result: result tensor, with shape [...]
+        :param X: input tensor, with shape [..., input_dim]
+        :param F: function evaluation tensor, with shape [..., latent_dim]
+        :param Y: observation tensor, with shape [..., observation_dim]
         """
         expected_shape = tf.broadcast_dynamic_shape(tf.shape(F)[:-1], tf.shape(Y)[:-1])
         tf.debugging.assert_equal(tf.shape(result), expected_shape)
 
+    def _check_input_dims(self, X: TensorType) -> None:
```

Comment:
I assume you will be adding your shape-checking functionality in a later PR?

Reply:
> I assume you will be adding your shape-checking functionality in a later PR?

Yes.